src/providers/github/Github.ts

```diff
@@ -69,18 +77,32 @@ export class Github {
    * @param tag "v0.2.0", "release/patch"
    */
   async deleteTagIfExists(tag: string): Promise<void> {
+    if (!(await this.tagExists(tag))) return;
+
+    await this.octokit.request("DELETE /repos/{owner}/{repo}/git/refs/{ref}", {
+      owner: this.owner,
+      repo: this.repo,
+      ref: `tags/${tag}`,
+      headers: {
+        "X-GitHub-Api-Version": "2022-11-28"
+      }
+    });
+  }
+
+  private async tagExists(tag: string): Promise<boolean> {
     try {
-      await this.octokit.git.deleteRef({
+      await this.octokit.request("GET /repos/{owner}/{repo}/git/ref/{ref}", {
         owner: this.owner,
         repo: this.repo,
-        ref: `tags/${tag}`
+        ref: `tags/${tag}`,
+        headers: {
+          "X-GitHub-Api-Version": "2022-11-28"
```

Comment:
this looks weird, why is this header required?

Reply:
It is not required, but it is a good practice, as it defines the version of the Github API, so that we ensure the response follows the expected format (in case there is a newer API version the request or response formats may vary)